api/src/models/FedoraObject.test.ts

```diff
@@ -47,6 +47,40 @@ describe("FedoraObject", () => {
             expect(spy).toHaveBeenCalledWith(expect.stringContaining("license1Url"), stream, "text/xml", [201, 204]);
         });
     });
+    describe("modifyAgents", () => {
+        let agents;
+        beforeEach(() => {
+            agents = [
+                {
+                    role: "test1",
+                    type: "test2",
+                    name: "test3",
+                    notes: ["test4"]
```

Comment:
To cover all edges, maybe it's worth also having a test with no notes, and a test with multiple notes.